sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -353,6 +355,18 @@ private CollectionCapacityConstraint GetInitializerConstraint(InitializerExpress
                     });
             }
 
+            private static bool HasAssociatedSymbolicValue(ArgumentSyntax argumentSyntax)
+            {
+                // e.g. Method(out var _)
+                if (DeclarationExpressionSyntaxWrapper.IsInstance(argumentSyntax.Expression) &&
+                    ((DeclarationExpressionSyntaxWrapper)argumentSyntax.Expression).Designation.SyntaxNode.IsKind(SyntaxKindEx.DiscardDesignation))
+                {
+                    return false;
+                }
+
+                return true;
+            }
```

Comment:
(personal preference) I'd use the sorthand version instead:
```
private static bool HasAssociatedSymbolicValue(ArgumentSyntax argumentSyntax) =>
    !DeclarationExpressionSyntaxWrapper.IsInstance(argumentSyntax.Expression) ||
      !((DeclarationExpressionSyntaxWrapper)argumentSyntax.Expression).Designation.SyntaxNode.IsKind(SyntaxKindEx.DiscardDesignation)  
```

Reply:
If it's fine with you I would leave it as it is since I find it a little bit harder to read due to the negation.